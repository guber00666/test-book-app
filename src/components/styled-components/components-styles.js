import styled from "styled-components";

export const StyledList = styled.div `
              margin-top: 30px;
              border: solid 1px;
              border-radius: 0.3em;
        `;
export const StyledPaginator = styled.div `
              margin-top: 30px;
              margin-bottom: 30px;
              float: right;
        `;
export const StyledApp = styled.div `
          margin-top: 30px;
    `;
export const StyledCriteria = styled.div `
             margin-top: 30px;
             font-weight: bold;
    `;
export const StyledForm = styled.div`
             border: solid 1px;
             border-radius: 0.3em;
    `;
export const StyledSearchPanel = styled.div `
    border: solid 1px;
    border-radius: 0.3em;
`;