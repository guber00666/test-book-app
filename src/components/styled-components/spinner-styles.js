import styled, {keyframes} from "styled-components";

export const SpinnerPulse  = styled.div `
                width: 200px;
                height: 200px;
                display: inline-block;
                overflow: hidden;
                background: #ffffff;
                margin-left: 26em;
        `;
const SpinnerElements1 = keyframes`
    0% { top: 39.199999999999996px; height: 121.60000000000001px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
const SpinnerElements2 = keyframes`
    0% { top: 44.89999999999999px; height: 110.20000000000002px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
const SpinnerElements3 = keyframes`
    0% { top: 50.6px; height: 98.8px }
    50% { top: 62px; height: 76px }
    100% { top: 62px; height: 76px }
`;
export const PulseElement = styled.div`
        width: 100%;
        height: 100%;
        position: relative;
        transform: translateZ(0) scale(1);
        backface-visibility: hidden;
        transform-origin: 0 0;
    `;
export const NthChild1 = styled.div `
         position: absolute; 
         width: 28px;
         box-sizing: content-box;
         left: 36px;
         background: #353535;
         animation: ${SpinnerElements1} 1s cubic-bezier(0,0.5,0.5,1) infinite;
         animation-delay: -0.2s
    `;
export const NthChild2 = styled.div `
          position: absolute; 
          width: 28px;
          box-sizing: content-box;
          left: 86px;
          background: #666666;
          animation: ${SpinnerElements2} 1s cubic-bezier(0,0.5,0.5,1) infinite;
          animation-delay: -0.1s
    `;
export const NthChild3 = styled.div `
           position: absolute; 
           width: 28px;
           box-sizing: content-box;
           left: 136px;
           background: #9b9b9b;
           animation: ${SpinnerElements3} 1s cubic-bezier(0,0.5,0.5,1) infinite;
           animation-delay: initial;
    `;