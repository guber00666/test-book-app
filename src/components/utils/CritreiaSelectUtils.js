import React from 'react';
import {criteriaOptions} from "../../constants/constants";

export const rowsOptions = criteriaOptions.map((el, id) => {
    return (
        <option key={id} value={el.value}>{el.label}</option>
    );
});