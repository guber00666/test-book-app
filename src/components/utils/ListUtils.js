import React from "react";

export const filterForElements = (books, filterValue, criteriaValue) => {
    let keyForFiter;
    switch (criteriaValue) {
        case ("1"):
            keyForFiter = 'genre';
            break;
        case ("2"):
            keyForFiter = 'author';
            break;
        default:
            keyForFiter = 'name';
    }
    return books.filter((item) => item[keyForFiter]
        .toLowerCase()
        .includes(filterValue.toLowerCase())
    );
};

