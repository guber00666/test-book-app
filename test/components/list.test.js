import React from "react";
import List from "../../src/components/list";
import Adapter from "enzyme-adapter-react-16/build";
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { StyledList, StyledPaginator } from "../../src/components/styled-components/components-styles";
import configureStore from "redux-mock-store";
import 'jest-styled-components';
import { filterForElements } from "../../src/components/utils/ListUtils";
import {
    mapStateToListProps,
    mapDispatchToListProps,
} from '../../src/actions/index';

configure({ adapter: new Adapter() });

export const initialState = {
    criteriaValue: '0',
    filterValue: '',
    paginatorValue: 1,
    books: [],
    isLoading: true,
    hasErrored: false
};

const middlewares = [];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);


describe("<List />", () => {

    const wrapperList = mount(
        <Provider store={store}>
            <List />
        </Provider>
    );

    it('should be render self and other import', () => {
        expect(wrapperList.find('div').at(1).hasClass('list')).toBe(true);
        expect(wrapperList.find('div').at(14).hasClass('pagination-container')).toBe(true);

        expect(wrapperList.find(StyledList)).toBeDefined();
        expect(wrapperList.find(StyledPaginator)).toBeDefined();

        expect(wrapperList).toMatchSnapshot();
    });

    it('List should have styles', () => {
        const listStyles = renderer.create(<StyledList />).toJSON();
        expect(listStyles).toHaveStyleRule('margin-top', '30px');
        expect(listStyles).toHaveStyleRule('border', 'solid 1px');
        expect(listStyles).toHaveStyleRule('border-radius', '0.3em');

        const paginatorStyles = renderer.create(<StyledPaginator />).toJSON();
        expect(paginatorStyles).toHaveStyleRule('margin-top', '30px');
        expect(paginatorStyles).toHaveStyleRule('margin-bottom', '30px');
        expect(paginatorStyles).toHaveStyleRule('float', 'right');

        expect(wrapperList).toMatchSnapshot();
    });

    it('List should show previously rolled value', () => {
        expect(mapStateToListProps(initialState).criteriaValue).toEqual("0");
        expect(mapStateToListProps(initialState).filterValue).toEqual("");
        expect(mapStateToListProps(initialState).paginatorValue).toEqual(1);
        expect(mapStateToListProps(initialState).books).toEqual([]);
        expect(mapStateToListProps(initialState).isLoading).toEqual(true);
        expect(mapStateToListProps(initialState).hasErrored).toEqual(false);
    });

    it('List should roll the dice again when onChange work out', () => {
        const dispatch = jest.fn();

        mapDispatchToListProps(dispatch).setPaginatorValue();
        expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SET_PAGE' });
    });

});

describe('filterForElements function', () => {
    it('should be filter by critieriaValue and filterValue', () => {
        const filter = filterForElements;
        const books = [
            {
                id: 1,
                name: 'Foundation',
                author: 'Asimov Isaac',
                genre: 'SciFi'
            },
            {
                id: 2,
                name: 'The Master and Margarita',
                author: 'Bulgakov Mikhail',
                genre: 'Fantasy'
            }
            ];
        const filterValue = 'SciFi';
        const criteriaValue = '1';
        expect(filter
        (books, filterValue, criteriaValue))
            .toEqual([{"author": "Asimov Isaac", "genre": "SciFi", "id": 1, "name": "Foundation"}]);
    });
});