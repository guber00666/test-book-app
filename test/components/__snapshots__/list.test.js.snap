// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> List should have styles 1`] = `
.c1 {
  width: 200px;
  height: 200px;
  display: inline-block;
  overflow: hidden;
  background: #ffffff;
  margin-left: 26em;
}

.c2 {
  width: 100%;
  height: 100%;
  position: relative;
  -webkit-transform: translateZ(0) scale(1);
  -ms-transform: translateZ(0) scale(1);
  transform: translateZ(0) scale(1);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c3 {
  position: absolute;
  width: 28px;
  box-sizing: content-box;
  left: 36px;
  background: #353535;
  -webkit-animation: cDTAEc 1s cubic-bezier(0,0.5,0.5,1) infinite;
  animation: cDTAEc 1s cubic-bezier(0,0.5,0.5,1) infinite;
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c4 {
  position: absolute;
  width: 28px;
  box-sizing: content-box;
  left: 86px;
  background: #666666;
  -webkit-animation: fOLAFE 1s cubic-bezier(0,0.5,0.5,1) infinite;
  animation: fOLAFE 1s cubic-bezier(0,0.5,0.5,1) infinite;
  -webkit-animation-delay: -0.1s;
  animation-delay: -0.1s;
}

.c5 {
  position: absolute;
  width: 28px;
  box-sizing: content-box;
  left: 136px;
  background: #9b9b9b;
  -webkit-animation: fEBSmt 1s cubic-bezier(0,0.5,0.5,1) infinite;
  animation: fEBSmt 1s cubic-bezier(0,0.5,0.5,1) infinite;
  -webkit-animation-delay: initial;
  animation-delay: initial;
}

.c0 {
  margin-top: 30px;
  border: solid 1px;
  border-radius: 0.3em;
}

.c6 {
  margin-top: 30px;
  margin-bottom: 30px;
  float: right;
}

<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(List)>
    <List
      books={Array []}
      criteriaValue="0"
      fetchData={[Function]}
      filterValue=""
      hasErrored={false}
      isLoading={true}
      paginatorValue={1}
      setPaginatorValue={[Function]}
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
              margin-top: 30px;
              border: solid 1px;
              border-radius: 0.3em;
        ",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <div
              className="list"
            >
              <Spinner>
                <div
                  className="spinner-container"
                >
                  <styled.div>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "c1",
                            "rules": Array [
                              "
             width: 200px;
             height: 200px;
             display: inline-block;
             overflow: hidden;
             background: #ffffff;
             margin-left: 26em;
        ",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c1"
                      >
                        <styled.div>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-bwzfXH",
                                  "isStatic": false,
                                  "lastClassName": "c2",
                                  "rules": Array [
                                    "
             width: 100%;
             height: 100%;
             position: relative;
             transform: translateZ(0) scale(1);
             backface-visibility: hidden;
             transform-origin: 0 0;
    ",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-bwzfXH",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c2"
                            >
                              <div
                                className="spinner-pulse"
                              >
                                <div
                                  className="spinner-elements"
                                >
                                  <styled.div>
                                    <StyledComponent
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-htpNat",
                                            "isStatic": false,
                                            "lastClassName": "c3",
                                            "rules": Array [
                                              "
             position: absolute; 
             width: 28px;
             box-sizing: content-box;
             left: 36px;
             background: #353535;
             animation: ",
                                              Keyframes {
                                                "id": "sc-keyframes-cDTAEc",
                                                "inject": [Function],
                                                "name": "cDTAEc",
                                                "rules": Array [
                                                  "@-webkit-keyframes cDTAEc{0%{top:39.199999999999996px;height:121.60000000000001px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                  "@keyframes cDTAEc{0%{top:39.199999999999996px;height:121.60000000000001px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                ],
                                                "toString": [Function],
                                              },
                                              " 1s cubic-bezier(0,0.5,0.5,1) infinite;
             animation-delay: -0.2s
    ",
                                            ],
                                          },
                                          "displayName": "styled.div",
                                          "foldedComponentIds": Array [],
                                          "render": [Function],
                                          "styledComponentId": "sc-htpNat",
                                          "target": "div",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                    >
                                      <div
                                        className="c3"
                                      >
                                        <div
                                          className="spinner-element-1"
                                        />
                                      </div>
                                    </StyledComponent>
                                  </styled.div>
                                  <styled.div>
                                    <StyledComponent
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-bxivhb",
                                            "isStatic": false,
                                            "lastClassName": "c4",
                                            "rules": Array [
                                              "
             position: absolute; 
             width: 28px;
             box-sizing: content-box;
             left: 86px;
             background: #666666;
             animation: ",
                                              Keyframes {
                                                "id": "sc-keyframes-fOLAFE",
                                                "inject": [Function],
                                                "name": "fOLAFE",
                                                "rules": Array [
                                                  "@-webkit-keyframes fOLAFE{0%{top:44.89999999999999px;height:110.20000000000002px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                  "@keyframes fOLAFE{0%{top:44.89999999999999px;height:110.20000000000002px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                ],
                                                "toString": [Function],
                                              },
                                              " 1s cubic-bezier(0,0.5,0.5,1) infinite;
             animation-delay: -0.1s
    ",
                                            ],
                                          },
                                          "displayName": "styled.div",
                                          "foldedComponentIds": Array [],
                                          "render": [Function],
                                          "styledComponentId": "sc-bxivhb",
                                          "target": "div",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                    >
                                      <div
                                        className="c4"
                                      >
                                        <div
                                          className="spinner-element-2"
                                        />
                                      </div>
                                    </StyledComponent>
                                  </styled.div>
                                  <styled.div>
                                    <StyledComponent
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-ifAKCX",
                                            "isStatic": false,
                                            "lastClassName": "c5",
                                            "rules": Array [
                                              "
             position: absolute; 
             width: 28px;
             box-sizing: content-box;
             left: 136px;
             background: #9b9b9b;
             animation: ",
                                              Keyframes {
                                                "id": "sc-keyframes-fEBSmt",
                                                "inject": [Function],
                                                "name": "fEBSmt",
                                                "rules": Array [
                                                  "@-webkit-keyframes fEBSmt{0%{top:50.6px;height:98.8px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                  "@keyframes fEBSmt{0%{top:50.6px;height:98.8px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                ],
                                                "toString": [Function],
                                              },
                                              " 1s cubic-bezier(0,0.5,0.5,1) infinite;
             animation-delay: initial;
    ",
                                            ],
                                          },
                                          "displayName": "styled.div",
                                          "foldedComponentIds": Array [],
                                          "render": [Function],
                                          "styledComponentId": "sc-ifAKCX",
                                          "target": "div",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                    >
                                      <div
                                        className="c5"
                                      >
                                        <div
                                          className="spinner-element-3"
                                        />
                                      </div>
                                    </StyledComponent>
                                  </styled.div>
                                </div>
                              </div>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </StyledComponent>
                  </styled.div>
                </div>
              </Spinner>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bZQynM",
                        "isStatic": false,
                        "lastClassName": "c6",
                        "rules": Array [
                          "
              margin-top: 30px;
              margin-bottom: 30px;
              float: right;
        ",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bZQynM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c6"
                  >
                    <div
                      className="pagination-container"
                    />
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </div>
        </StyledComponent>
      </styled.div>
    </List>
  </Connect(List)>
</Provider>
`;

exports[`<List /> should be render self and other import 1`] = `
.c1 {
  width: 200px;
  height: 200px;
  display: inline-block;
  overflow: hidden;
  background: #ffffff;
  margin-left: 26em;
}

.c2 {
  width: 100%;
  height: 100%;
  position: relative;
  -webkit-transform: translateZ(0) scale(1);
  -ms-transform: translateZ(0) scale(1);
  transform: translateZ(0) scale(1);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c3 {
  position: absolute;
  width: 28px;
  box-sizing: content-box;
  left: 36px;
  background: #353535;
  -webkit-animation: cDTAEc 1s cubic-bezier(0,0.5,0.5,1) infinite;
  animation: cDTAEc 1s cubic-bezier(0,0.5,0.5,1) infinite;
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c4 {
  position: absolute;
  width: 28px;
  box-sizing: content-box;
  left: 86px;
  background: #666666;
  -webkit-animation: fOLAFE 1s cubic-bezier(0,0.5,0.5,1) infinite;
  animation: fOLAFE 1s cubic-bezier(0,0.5,0.5,1) infinite;
  -webkit-animation-delay: -0.1s;
  animation-delay: -0.1s;
}

.c5 {
  position: absolute;
  width: 28px;
  box-sizing: content-box;
  left: 136px;
  background: #9b9b9b;
  -webkit-animation: fEBSmt 1s cubic-bezier(0,0.5,0.5,1) infinite;
  animation: fEBSmt 1s cubic-bezier(0,0.5,0.5,1) infinite;
  -webkit-animation-delay: initial;
  animation-delay: initial;
}

.c0 {
  margin-top: 30px;
  border: solid 1px;
  border-radius: 0.3em;
}

.c6 {
  margin-top: 30px;
  margin-bottom: 30px;
  float: right;
}

<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(List)>
    <List
      books={Array []}
      criteriaValue="0"
      fetchData={[Function]}
      filterValue=""
      hasErrored={false}
      isLoading={true}
      paginatorValue={1}
      setPaginatorValue={[Function]}
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
              margin-top: 30px;
              border: solid 1px;
              border-radius: 0.3em;
        ",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <div
              className="list"
            >
              <Spinner>
                <div
                  className="spinner-container"
                >
                  <styled.div>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "c1",
                            "rules": Array [
                              "
             width: 200px;
             height: 200px;
             display: inline-block;
             overflow: hidden;
             background: #ffffff;
             margin-left: 26em;
        ",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="c1"
                      >
                        <styled.div>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-bwzfXH",
                                  "isStatic": false,
                                  "lastClassName": "c2",
                                  "rules": Array [
                                    "
             width: 100%;
             height: 100%;
             position: relative;
             transform: translateZ(0) scale(1);
             backface-visibility: hidden;
             transform-origin: 0 0;
    ",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-bwzfXH",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c2"
                            >
                              <div
                                className="spinner-pulse"
                              >
                                <div
                                  className="spinner-elements"
                                >
                                  <styled.div>
                                    <StyledComponent
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-htpNat",
                                            "isStatic": false,
                                            "lastClassName": "c3",
                                            "rules": Array [
                                              "
             position: absolute; 
             width: 28px;
             box-sizing: content-box;
             left: 36px;
             background: #353535;
             animation: ",
                                              Keyframes {
                                                "id": "sc-keyframes-cDTAEc",
                                                "inject": [Function],
                                                "name": "cDTAEc",
                                                "rules": Array [
                                                  "@-webkit-keyframes cDTAEc{0%{top:39.199999999999996px;height:121.60000000000001px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                  "@keyframes cDTAEc{0%{top:39.199999999999996px;height:121.60000000000001px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                ],
                                                "toString": [Function],
                                              },
                                              " 1s cubic-bezier(0,0.5,0.5,1) infinite;
             animation-delay: -0.2s
    ",
                                            ],
                                          },
                                          "displayName": "styled.div",
                                          "foldedComponentIds": Array [],
                                          "render": [Function],
                                          "styledComponentId": "sc-htpNat",
                                          "target": "div",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                    >
                                      <div
                                        className="c3"
                                      >
                                        <div
                                          className="spinner-element-1"
                                        />
                                      </div>
                                    </StyledComponent>
                                  </styled.div>
                                  <styled.div>
                                    <StyledComponent
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-bxivhb",
                                            "isStatic": false,
                                            "lastClassName": "c4",
                                            "rules": Array [
                                              "
             position: absolute; 
             width: 28px;
             box-sizing: content-box;
             left: 86px;
             background: #666666;
             animation: ",
                                              Keyframes {
                                                "id": "sc-keyframes-fOLAFE",
                                                "inject": [Function],
                                                "name": "fOLAFE",
                                                "rules": Array [
                                                  "@-webkit-keyframes fOLAFE{0%{top:44.89999999999999px;height:110.20000000000002px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                  "@keyframes fOLAFE{0%{top:44.89999999999999px;height:110.20000000000002px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                ],
                                                "toString": [Function],
                                              },
                                              " 1s cubic-bezier(0,0.5,0.5,1) infinite;
             animation-delay: -0.1s
    ",
                                            ],
                                          },
                                          "displayName": "styled.div",
                                          "foldedComponentIds": Array [],
                                          "render": [Function],
                                          "styledComponentId": "sc-bxivhb",
                                          "target": "div",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                    >
                                      <div
                                        className="c4"
                                      >
                                        <div
                                          className="spinner-element-2"
                                        />
                                      </div>
                                    </StyledComponent>
                                  </styled.div>
                                  <styled.div>
                                    <StyledComponent
                                      forwardedComponent={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "attrs": Array [],
                                          "componentStyle": ComponentStyle {
                                            "componentId": "sc-ifAKCX",
                                            "isStatic": false,
                                            "lastClassName": "c5",
                                            "rules": Array [
                                              "
             position: absolute; 
             width: 28px;
             box-sizing: content-box;
             left: 136px;
             background: #9b9b9b;
             animation: ",
                                              Keyframes {
                                                "id": "sc-keyframes-fEBSmt",
                                                "inject": [Function],
                                                "name": "fEBSmt",
                                                "rules": Array [
                                                  "@-webkit-keyframes fEBSmt{0%{top:50.6px;height:98.8px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                  "@keyframes fEBSmt{0%{top:50.6px;height:98.8px;}50%{top:62px;height:76px;}100%{top:62px;height:76px;}}",
                                                ],
                                                "toString": [Function],
                                              },
                                              " 1s cubic-bezier(0,0.5,0.5,1) infinite;
             animation-delay: initial;
    ",
                                            ],
                                          },
                                          "displayName": "styled.div",
                                          "foldedComponentIds": Array [],
                                          "render": [Function],
                                          "styledComponentId": "sc-ifAKCX",
                                          "target": "div",
                                          "toString": [Function],
                                          "warnTooManyClasses": [Function],
                                          "withComponent": [Function],
                                        }
                                      }
                                      forwardedRef={null}
                                    >
                                      <div
                                        className="c5"
                                      >
                                        <div
                                          className="spinner-element-3"
                                        />
                                      </div>
                                    </StyledComponent>
                                  </styled.div>
                                </div>
                              </div>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </StyledComponent>
                  </styled.div>
                </div>
              </Spinner>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bZQynM",
                        "isStatic": false,
                        "lastClassName": "c6",
                        "rules": Array [
                          "
              margin-top: 30px;
              margin-bottom: 30px;
              float: right;
        ",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bZQynM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c6"
                  >
                    <div
                      className="pagination-container"
                    />
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </div>
        </StyledComponent>
      </styled.div>
    </List>
  </Connect(List)>
</Provider>
`;
